docs/LabVIEW_Debugging_Techniques.md

```diff
@@ -0,0 +1,128 @@
+Once added to the system definition, custom devices have been fully integrated into VeriStand’s context. As a result, LabVIEW’s built-in debugging techniques are no longer available. Several techniques are available for debugging and benchmarking the custom device.
+ 
+#### Console Viewer
+
+A subcomponent of VeriStand RT Engine is the RT Console Viewer.
+You can install it to the execution host using Measurement and Automation Explorer. When installed, the component runs a small UDP daemon allowing the operator to view the console from a utility called the RT Console Viewer. You can access the RT Console viewer from **VeriStand » Workspace » Tools » Console Viewer**.	
+
+![](images/ConsoleViewer1.jpg)
+
+**Figure: RT Console Viewer**
+
+The Console Viewer will show the system definition and the resulting CPU usage. The viewer is also useful for displaying debugging messages. The console viewer provides a periodic snapshot of utilization. CPU spikes and transients will probably be unobservable. If the system is very busy, it may not update the console viewer at all. You can use other debugging methods for a more accurate indication of resource utilization.
+
+As the name implies, the RT Console viewer is only available on real-time targets. The RT Console Viewer is also available as a stand-alone add-on to LabVIEW Real-Time. See **[NI Developer Zone Tutorial: Remotely View Console Output of Real-Time Targets](https://knowledge.ni.com/KnowledgeArticleDetails?id=kA03q000000x4TjCAI&l=en-US)** for more information.
+
+#### Printing to the Console
+Printing to the console is often all that’s needed to debug an application.
+
+#### Printing With NIVS Debug String VI
+The recommended method of printing to the console is to use Print NIVS Debug String VI. You can download the VI from NI Community » VeriStand Add-Ons » Documents » Print VeriStand Debug String.
+
+![](images/NIVSdebugString.jpg)
+
+This VI works on both Windows and RT execution hosts. It has an optional input to change the color of the text. It also has an optional input to append the string to the VeriStand log file.
+
+#### Printing With ni_emb.dll
+
+The NIVS Debug String VI is not available in VeriStand 2009. You’ll find **ni_emb.dll** in
+**<labview>\Targets\NI\RT\vi.lib**. This dll contains a stub function called **PrintStringToConsole**. Calling this function sends a string to the RT console. Configure the function to run in any thread using the C calling convention. The return type is void and it has a C String pointer input constant. You’ll find this function wrapped in a VI in the same folder in **rtutility.llb\RT Debug String.vi**. Since **ni_emb.dll** is a stub dll, it’s not necessary to deploy this VI to the RT target. The stub exists so the **PrintStringToConsole** function does not return an error when called on Windows.
```

Comment:
```suggestion
**&lt;labview&gt;\Targets\NI\RT\vi.lib**. This dll contains a stub function called **PrintStringToConsole**. Calling this function sends a string to the RT console. Configure the function to run in any thread using the C calling convention. The return type is void and it has a C String pointer input constant. You’ll find this function wrapped in a VI in the same folder in **rtutility.llb\RT Debug String.vi**. Since **ni_emb.dll** is a stub dll, it’s not necessary to deploy this VI to the RT target. The stub exists so the **PrintStringToConsole** function does not return an error when called on Windows.
```